py3status/modules/xkb_input.py

```diff
@@ -0,0 +1,507 @@
+# -*- coding: utf-8 -*-
+"""
+Switch inputs.
+
+Configuration parameters:
+    button_next: mouse button to cycle next layout (default 4)
+    button_prev: mouse button to cycle previous layout (default 5)
+    cache_timeout: refresh interval for this module; xkb-switch
+        and swaymsg will listen for new updates instead (default 10)
+    format: display format for this module (default '{format_input}')
+    format_input: display format for inputs
+        (default '[{alias}][\?soft  ][\?color=s {s}[ {v}]]')
+    format_input_separator: show separator if more than one (default ' ')
+    format_libinput: display format for libinputs (default None)
+    inputs: specify a list of inputs to use in swaymsg (default [])
+    switcher: specify xkb-switch, xkblayout-state, xkbgroup,
+        or swaymsg to use, otherwise auto (default None)
+    thresholds: specify color thresholds to use
+        *(default [("fr", "lightgreen"), ("ru", "lightcoral"),
+        ("ua", "khaki"),  ("us", "lightskyblue")])*
+
+Format placeholders:
+    {format_input} format for inputs
+    {input}        number of inputs, eg 1
+    {switcher}     eg, xkb-switch, xkblayout-state, xkbgroup, swaymsg
+
+format_input placeholders:
+    xkb-switch:
+    xkblayout-state:
+    xkbgroup:
+    swaymsg:
+        {c} layout number, eg, 0
+        {n} layout name, eg, English (US)
+        {s} layout symbol, eg, us
+        {v} layout variant, eg, basic
+        {e} layout variant, {v} or {s}, eg, dvorak
+        {C} layout count, eg, 2
+    swaymsg:
+        {alias}                   custom string or {name}
+        {identifier}              eg, 162:253 USB-HID Keyboard
+        {name}                    eg, Trackball, Keyboard, etc
+        {vendor}                  eg, 320
+        {product}                 eg, 556
+        {type}                    eg, pointer, keyboard, touchpad, etc
+        {xkb_layout_names}        eg, English (US), French, Russian
+        {xkb_active_layout_index} eg, 0, 1, 2, etc
+        {xkb_active_layout_name}  eg, English (US)
+        {format_libinput}         format for libinputs
+
+format_libinput placeholders:
+    {send_events}      eg, enabled
+    {accel_speed}      eg, 0.0
+    {accel_profile}    eg, adaptive
+    {natural_scroll}   eg, adaptive
+    {left_handed}      eg, disabled
+    {middle_emulation} eg, disabled
+    {scroll_method}    eg, none
+    {scroll_button}    eg, 274
```

Comment:
I went all out. Should this be included or not? It may be better to swap `enabled, disabled, none` with `True, False, None` and use `self.py3.flatten_dict` instead... i.e. `\?if=send_events {alias}` to hide disabled inputs should some users use certain controls to change certain libinputs... or to change output slightly because of `left_handed`.... such as `\?if=left_handed Left {alias}|{alias} Right`

I assume almost no chance of users using things like this.